### DISCUSSION TITLE
Dell U2520D volume control is not working

### DISCUSSION THREAD

**Discussion by LightmoonXD** [MonitorControl/MonitorControl#1122] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Volume control isnt working while seeing the volume variation via native macos volume panel.i am sure DDC is working well.
The brightness control is fine.
Mac->U2520D->external speaker with USB and 3.5mm audio output

### Expected behavior

Adjusting volume.

### Environment Information (please complete the following information)

```markdown
- macOS version:11.4 Big Sur
- Mac model:Macbook Pro (15-inch,mid 2014)
- MonitorControl version:4.1.0 version 7034
- Monitor(s):DELL U2520D
- Monitor Cable(s)/Connection(s):Mac->DP2.1->DELL U2520D
- Apple Silicon/M1 (yes or no):no
```

**Comment by 1P1E10** [comment]
Doesn't work for my Benq PD2705U neither.

**Comment by myzinsky** [comment]
Same here with DELL U2722DE

**Comment by xmollv** [comment]
I've the same issue with a DELL U2723QE. Brightness works fine, audio doesn't.

**Reply by txoof** [reply]
Similar issue with DELL U2723QE:

- Brightness works as expected, volume control does not work

**Comment by waydabber** [comment]
Many displays support brightness control only via DDC and no volume control. My display (Lenovo Creator Extreme) also has no volume controls via DDC. So I guess I'll join the list. ;)

**Comment by myzinsky** [comment]
So @tucknology's work-around is great. If that would be integrated also in the monitorControl tool that would be really handy!

**Comment by Fudal** [comment]
I've the same issue with a Dell S2721H. Brightness works fine, audio doesn't. Connected through HDMI.

**Comment by 1P1E10** [comment]
> I've the same issue with a Dell S2721H. Brightness works fine, audio doesn't. Connected through HDMI.

I connected via type c

**Comment by waydabber** [comment]
Audio control via HDMI will not work as HDMI on Mac blocks DDC. Otherwise these are possibly display firmware issues.

I'll move this thread over to discussions so everybody can chime in with their display.

**Reply by Fudal** [reply]
@waydabber if I connect my external display to Macbook via USB-C to HDMI adapter will it work?

**Reply by waydabber** [reply]
I am not sure. Most USB-C to HDMI adapters do relay DDC commands properly. However if your display do not support audio control via DDC, then fixing the connection will not solve the issue (but on the other hand might make backlight control work).

**Comment by krystof-k** [comment]
I have two Dell UP2516Ds connected via USB-C–DisplayPort, which have only 3,5 mm jack analog output, no speakers. Which is the reason it is not possible to control the volume I think – the monitor is simply not capable of doing that, there is no volume control in the OSD menu as well.

The only workaround I found working is [eqMac](https://eqmac.app/) app. It simply creates virtual sound device which then redirects output to the monitor. And supports volume control. Working all right so far.

**Reply by waydabber** [reply]
yes, eqMac or other virtual audio devices that act a filter-like way with digital volume control will give a similar effect. If an external speaker is connected to a display, the display OSD (or DDC control) would not be able to adjust the external speaker's amplifier anyway, only the line level of the display (if the display has support for it) so using eqMac or other similar software is perfectly fine and has a similar end result. 

(with a proper network controllable amplifier, you can however use some hacks to control that directly with the native apple keyboard volume controls - like I do with my Yamaha AV amplifier which I use with my Mac -, this allows for undisturbed digital audio all the way to the receiver and comfortable volume control at the same time.)
